api/v0/client/libp2p/client.go

```diff
@@ -0,0 +1,76 @@
+package p2pclient
+
+import (
+	"context"
+	"fmt"
+
+	v0 "github.com/filecoin-project/indexer-reference-provider/api/v0"
+	"github.com/filecoin-project/indexer-reference-provider/api/v0/models"
+	pb "github.com/filecoin-project/indexer-reference-provider/api/v0/pb"
+	"github.com/filecoin-project/indexer-reference-provider/internal/libp2pclient"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type Provider struct {
+	p2pc *libp2pclient.Client
+}
+
+func NewProvider(ctx context.Context, h host.Host, options ...libp2pclient.ClientOption) (*Provider, error) {
```

Comment:
Subjective comment: I wonder if having the word `Client` in the name of `Provider` client interface would read better from the perspective of developers using this library.

Reply:
You're gonna laugh, but I stared at this definition wondering the same for a few minutes. I settled for `NewProvider` to make it shorter. But if you are also wondering this, it means it has to be `NewProviderClient`. Thanks!